content/docs/querying/functions.md

```diff
@@ -159,6 +159,29 @@ increase(http_requests_total{job="api-server"}[5m])
 human readability. Use `rate` in recording rules so that increases are tracked
 consistently on a per-second basis.
 
+## `irate()`
+
+`irate(v range-vector)` calculates the per-second instant rate of increase of
+the time series in the range vector. This is based on the last two data points.
+Breaks in monotonicity (such as counter resets due to target restarts) are
+automatically adjusted for.
+
+The following example expression returns the per-second rate of HTTP requests
+looking up to 5 minutes back for the two most recent data points, per time
+series in the range vector:
+
+```
+irate(http_requests_total{job="api-server"}[5m])
+```
+
+`irate` should only be used when graphing volatile fast-moving counters.
+Use `rate` for alerts and slow-moving counters.
```

Comment:
Even though those function specs should be short and precise, a brief reasoning would certainly be helpful here.